commit c847d47cb7b2fa78b17c9e17ed3fbd010ee3d3ca
Author: Andrew Morton <akpm@linux-foundation.org>
Date:   Fri Jan 2 13:50:07 2009 +0000

    drivers/char/cyclades.c: cy_pci_probe: fix error path
    
    We forgot to release resources in one case.
    
    Addresses http://bugzilla.kernel.org/show_bug.cgi?id=12137
    
    Reported-by: Florian Lohoff <flo@rfc822.org>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Alan Cox <alan@redhat.com>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

diff --git a/drivers/char/cyclades.c b/drivers/char/cyclades.c
index 5e5b1dc..6a59f72 100644
--- a/drivers/char/cyclades.c
+++ b/drivers/char/cyclades.c
@@ -5010,7 +5010,7 @@ static int __devinit cy_pci_probe(struct pci_dev *pdev,
 		if (nchan == 0) {
 			dev_err(&pdev->dev, "Cyclom-Y PCI host card with no "
 					"Serial-Modules\n");
-			return -EIO;
+			goto err_unmap;
 		}
 	} else if (device_id == PCI_DEVICE_ID_CYCLOM_Z_Hi) {
 		struct RUNTIME_9060 __iomem *ctl_addr;
